packages/react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -539,276 +528,267 @@ const ComboBox = forwardRef(
-                if (isOpen) {
-                  toggleMenu();
-                }
-              }
-            },
-          });
-
-          const handleFocus = (evt: FocusEvent<HTMLDivElement>) => {
-            setIsFocused(evt.type === 'focus');
-          };
-
-          const readOnlyEventHandlers = readOnly
-            ? {
-                onKeyDown: (evt: KeyboardEvent<HTMLInputElement>) => {
-                  // This prevents the select from opening for the above keys
-                  if (evt.key !== 'Tab') {
-                    evt.preventDefault();
+                  if (match(event, keys.End) && event.code !== 'Numpad1') {
+                    event.target.setSelectionRange(
+                      event.target.value.length,
+                      event.target.value.length
+                    );
+                  }
+
+                  if (event.altKey && event.key == 'ArrowDown') {
+                    event.preventDownshiftDefault = true;
+                    if (!isOpen) {
+                      toggleMenu();
+                    }
+                  }
+                  if (event.altKey && event.key == 'ArrowUp') {
+                    event.preventDownshiftDefault = true;
+                    if (isOpen) {
+                      toggleMenu();
+                    }
                   }
                 },
-              }
-            : {};
-
-          // The input should be described by the appropriate message text id
-          // when both the message is supplied *and* when the component is in
-          // the matching state (invalid, warn, etc).
-          const ariaDescribedBy =
-            (invalid && invalidText && invalidTextId) ||
-            (warn && warnText && warnTextId) ||
-            (helperText && !isFluid && helperTextId) ||
-            undefined;
-
-          return (
-            <div className={wrapperClasses}>
-              {titleText && (
-                <Text as="label" className={titleClasses} {...labelProps}>
-                  {titleText}
-                </Text>
-              )}
-              <ListBox
-                onFocus={handleFocus}
-                onBlur={handleFocus}
-                className={className}
-                disabled={disabled}
-                invalid={invalid}
-                invalidText={invalidText}
-                invalidTextId={invalidTextId}
-                isOpen={isOpen}
-                light={light}
-                size={size}
-                warn={warn}
-                warnText={warnText}
-                warnTextId={warnTextId}>
-                <div className={`${prefix}--list-box__field`}>
-                  <input
-                    role="combobox"
-                    disabled={disabled}
-                    className={inputClasses}
-                    type="text"
-                    tabIndex={0}
-                    aria-autocomplete="list"
-                    aria-expanded={rootProps['aria-expanded']}
```

Comment:
By removing this `aria-expanded` attr, might affect this test file _[test-helpers.js](https://github.com/carbon-design-system/carbon/blob/7b9a05be22b9bee34f70299f7568f1aeb6e7c06a/packages/react/src/components/ListBox/test-helpers.js#L57)_, right?

```
export const findPopupNode = () => {
  return document.querySelector('[aria-haspopup="listbox"]');
};

expect(findPopupNode()).toHaveAttribute('aria-expanded', 'true');
```